cmd/talosctl/cmd/talos/kubeconfig.go

```diff
@@ -72,7 +84,59 @@ Kubeconfig will be written to PWD/kubeconfig or [local-path]/kubeconfig if speci
 	},
 }
 
+func extractAndMerge(args []string, r io.ReadCloser) error {
+	data, err := helpers.ExtractFileFromTarGz("kubeconfig", r)
+	if err != nil {
+		return err
+	}
+
+	var localPath string
+
+	if len(args) == 1 {
+		localPath = filepath.Clean(args[0])
+	} else {
+		var usr *user.User
+		usr, err = user.Current()
+
+		if err != nil {
+			return err
+		}
+
+		localPath = filepath.Join(usr.HomeDir, ".kube/config")
+	}
+
+	config, err := clientcmd.Load(data)
+	if err != nil {
+		return err
+	}
+
+	// base file does not exist, dump config as is
+	if _, err = os.Stat(localPath); os.IsNotExist(err) {
+		err = os.MkdirAll(filepath.Dir(localPath), 0644)
```

Comment:
as it's a directory, we should create it with 0777 and umask should take care of the rest? (or at least 755).